src/plus/integrations/authentication/integrationAuthentication.ts

```diff
@@ -154,42 +312,43 @@ export class IntegrationAuthenticationService implements Disposable {
 		}
 	}
 
-	private getSecretKey(providerId: IntegrationId, id: string): `gitlens.integration.auth:${IntegrationId}|${string}` {
-		return `gitlens.integration.auth:${providerId}|${id}`;
-	}
-
 	private async ensureProvider(providerId: IntegrationId): Promise<IntegrationAuthenticationProvider> {
 		let provider = this.providers.get(providerId);
 		if (provider == null) {
 			switch (providerId) {
 				case HostingIntegrationId.AzureDevOps:
 					provider = new (
 						await import(/* webpackChunkName: "integrations" */ './azureDevOps')
-					).AzureDevOpsAuthenticationProvider();
+					).AzureDevOpsAuthenticationProvider(this.container);
 					break;
 				case HostingIntegrationId.Bitbucket:
 					provider = new (
 						await import(/* webpackChunkName: "integrations" */ './bitbucket')
-					).BitbucketAuthenticationProvider();
+					).BitbucketAuthenticationProvider(this.container);
+					break;
+				case HostingIntegrationId.GitHub:
+					provider = new (
+						await import(/* webpackChunkName: "integrations" */ './github')
+					).GitHubAuthenticationProvider(this.container);
 					break;
 				case SelfHostedIntegrationId.GitHubEnterprise:
 					provider = new (
 						await import(/* webpackChunkName: "integrations" */ './github')
-					).GitHubEnterpriseAuthenticationProvider();
+					).GitHubEnterpriseAuthenticationProvider(this.container);
 					break;
 				case HostingIntegrationId.GitLab:
 				case SelfHostedIntegrationId.GitLabSelfHosted:
 					provider = new (
 						await import(/* webpackChunkName: "integrations" */ './gitlab')
-					).GitLabAuthenticationProvider();
+					).GitLabAuthenticationProvider(this.container, providerId);
```

Comment:
We should no longer need to pass `providerId` here.

Reply:
After GITCL-555 yes. But now it can be GitLabSelfHosted or GitLab

Reply:
Ah I see, good call.